django/db/backends/base/introspection.py

```diff
@@ -80,7 +80,10 @@ def django_table_names(self, only_existing=False, include_views=True):
                 if not model._meta.managed:
                     continue
                 tables.add(model._meta.db_table)
-                tables.update(f.m2m_db_table() for f in model._meta.local_many_to_many)
+                tables.update(f.m2m_db_table() for f in model._meta.local_many_to_many
+                              if not f.remote_field or
+                              not f.remote_field.through or
+                              f.remote_field.through._meta.managed)
```

Comment:
Ah, okay. I originally wrote the patch on 1.4 (don't ask), which has f.rel: which wasn't always set (or f.rel.through wasn't always there).

So I still need the `if not f.remote_field.through` though, because that's only present when there is an explicitly defined through model, right?